applications/rag/frontend/container/rai/nlp_filter.py

```diff
@@ -0,0 +1,72 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+from google.cloud import language_v1
+from . import retry
+
+# Convert the project id into a full resource id.
+parent = os.environ.get('PROJECT_ID', 'NULL')
+
+# Instantiate a nlp client.
+nature_language_client = language_v1.LanguageServiceClient()
+
+def is_nlp_api_enabled():
+  if parent == 'NULL':
+    return False
+  # Check if the DLP API is enabled
+  try:
+    sum_confidences("test")
+    return True
+  except Exception as e:
+    print(f"Error: {e}")
+    return False
+
+def sum_confidences(text):
+  document = language_v1.types.Document(
+      content=text, type_=language_v1.types.Document.Type.PLAIN_TEXT
+  )
+
+  request = language_v1.ModerateTextRequest(
+      document=document,
+  )
+  # Detects the sentiment of the text
+  response = nature_language_client.moderate_text(
+      request=request, retry=retry.retry_policy
+  )
+
+  # Parse response and sum the confidences
+  total_confidence = 0.0
+  categories = []
+  for category in response.moderation_categories:
+    total_confidence += category.confidence
+    categories.append(category.name)
+  return total_confidence, categories
+
+def is_content_inappropriate(text, nlp_filter_level):
+  # Define thresholds
```

Comment:
Is there a grounding for these numbers? They seem somewhat magic.

Reply:
No, it is based on my experience. 
Some example is:
Cons:Beds were VERY uncomfortable.  I Could hear the neighbors every word and anytime someone would use water, the doors were constantly slamming shut all night. The hotel was very confusing to navigate with all the different buildings and walkways.

Pros:Room was clean, midway arcade was fun for kids. Slide and pool were fun.
Respo: moderation_categories {
  name: "Toxic"
  confidence: 0.0167134274
}
moderation_categories {
  name: "Insult"
  confidence: 0.0144531988
}
moderation_categories {
  name: "Profanity"
  confidence: 0.00326660462
}
moderation_categories {
  name: "Derogatory"
  confidence: 0.00357374363
}
moderation_categories {
  name: "Sexual"
  confidence: 0.00243523018
}
moderation_categories {
  name: "Death, Harm & Tragedy"
  confidence: 0.189759031
}
moderation_categories {
  name: "Violent"
  confidence: 0.028125
}
moderation_categories {
  name: "Firearms & Weapons"
}
moderation_categories {
  name: "Public Safety"
  confidence: 0.140845075
}
moderation_categories {
  name: "Health"
  confidence: 0.359649122
}
moderation_categories {
  name: "Religion & Belief"
  confidence: 0.055555556
}
moderation_categories {
  name: "Illicit Drugs"
  confidence: 0.0573770478
}
moderation_categories {
  name: "War & Conflict"
  confidence: 0.121951222
}
moderation_categories {
  name: "Politics"
  confidence: 0.034608379
}
moderation_categories {
  name: "Finance"
  confidence: 0.101694912
}
moderation_categories {
  name: "Legal"
  confidence: 0.100946374
}

Reply:
Discussed offline the magic number will be adjusted a little bit after user testing